File: backend/src/main/java/com/paperai/model/entity/CustomAgent.java
package com.paperai.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("custom_agent")
public class CustomAgent implements java.io.Serializable {
    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    /** 角色名，如"数据分析专家" */
    private String name;

    /** 图标 emoji */
    private String icon;

    /** 角色描述 */
    private String description;

    /** 自定义 System Prompt */
    private String systemPrompt;

    /** 默认模型 */
    private String model;

    /** 默认温度 */
    private Double temperature;

    /** 是否启用 */
    private Integer enabled;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: backend-py/app/models/custom_agent.py
from sqlalchemy import String, Text, Integer, BigInteger, Float, Index
from sqlalchemy.orm import Mapped, mapped_column
from .base import Base, TimestampMixin


class CustomAgent(Base, TimestampMixin):
    __tablename__ = "custom_agent"
    __table_args__ = (
        Index("ix_ca_user_id", "user_id"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    icon: Mapped[str] = mapped_column(String(10), default="🤖")
    description: Mapped[str | None] = mapped_column(String(500))
    system_prompt: Mapped[str | None] = mapped_column(Text)
    model: Mapped[str] = mapped_column(String(100), nullable=False, default="mimo-v2.5-pro")
    temperature: Mapped[float] = mapped_column(Float, nullable=False, default=0.7)
    enabled: Mapped[int] = mapped_column(Integer, nullable=False, default=1)
